cookbook/core/extend_flyte/container_interface.py

```diff
@@ -0,0 +1,76 @@
+"""
+.. _core-extend-flyte-container-interface:
+
+Container Interface
+-------------------
+The Flyte typically interacts with containers in the course of its task execution (since most tasks are container
+tasks). If you are extending Flyte with a new container task type, it's worth it to understand how the platform
+interacts with containers. This interaction is encoded in two places.
+
+#. At compilation time for a container task, the arguments to that container are set. This is done by flytekit
+   for instance for your run of the mill ``@task``.
+#. At runtime, Flyte will execute your task via a plugin. The default container plugin (and most other plugins but
+   not all) will do the following:
+  #. Set a series of environment variables.
+  #. Before running the container, the plugin will search/replace values in the container arguments. The
+     command templating section below details how this happens.
+
+Command Templating
+^^^^^^^^^^^^^^^^^^
+The templating of container arguments at run-time is one of the more advanced constructs of Flyte, but one that
+authors of new task types should be aware of. For example, when looking at the hello world task in the UI,
+if you click the Task tab, you'd see JSON that contains something like the following:
+
+.. code-block:: json
+
+   "container": {
+     "command": [],
+     "args": [
+       "pyflyte-execute",
+       "--inputs",
+       "{{.input}}",
+       "--output-prefix",
+       "{{.outputPrefix}}",
+       "--raw-output-data-prefix",
+       "{{.rawOutputDataPrefix}}",
+       "--resolver",
+       "flytekit.core.python_auto_container.default_task_resolver",
+       "--",
+       "task-module",
+       "core.basic.hello_world",
+       "task-name",
+       "say_hello"
+     ],
+
+The following table explains what each of the ``{{}}`` items mean, along with some others.
+
+ +--------------------------+-----------------------------+----------------------------------------------------------+
+ |       Template           |        Example              |                  Description                             |
+ +==========================+=============================+==========================================================+
+ |  {{.Input}}              | ``s3://my-bucket/inputs.pb``| Pb file containing a LiteralMap containing the inputs    |
+ +--------------------------+-----------------------------+----------------------------------------------------------+
+ |  {{.InputPrefix}}        | ``s3://my-bucket``          | Just the bucket where the inputs.pb file can be found    |
```

Comment:
to be clear, is this just the bucket or can it be the directory prefix? the template name itself is ambiguous

Reply:
this does not need to be containers. it can be for any task right that these are substitutable - for queries too